index.html

```diff
@@ -259,10 +259,198 @@ <h1>Architecture</h1>
+	In both cases, a Thing Link resolves a single URL into one or more additional URLs
+	which are fed back into the Introduction set, replacing the original URL pointing 
+	at the Thing Link.  A Thing Description Directory can also store Thing Links
+	which can redirect to other Thing Description Directories, allowing for a linked
+	directory structure.
+	</li>
+	</ul>
+    </section>
+    <section id="consumer-process" class="normative">
+        <h1>Discovering Clients and Consumers</h1>
+	<p>In this section we will describe the WoT Discovery process from the point of view of a client,
+	and what it means to say that a client supports WoT Discovery.
+	We will define a Discovering Client, D-Client for short, as an entity that supports the WoT Discovery
+	process.  Such a client may or may not be a Consumer.  A D-Client finds TDs, which are then passed
+	onto a Consumer for interpretation.  A D-Client itself may do some interpretation of TDs that it
+	fetches, however, in particular to examine the types of the TD it fetches to see if additional
+	fetches are required.  A Consumer may also be D-Client, in which case we will call
+	it a Discovering Consumer, or D-Consumer.
+	All of the statements below about D-Clients also apply to D-Consumers.
+
+	These requirements are designed so that nearly any client that can fetch a single TD 
+	given a single URL over any protocol (including "file") can be said to support WoT Discovery. 
+	Of course, clients may support more powerful Discovery mechanisms, and some of these 
+	have their own requirements.  
+	For example, using WoT Thing Description Directories (TDDs) requires the use of HTTP, and so
+	are only available to clients that support this protocol.  However, use of HTTP (and Directories)
+	is not in fact a requirement of the WoT Discovery process: it is possible to discover multiple
+	TDs using just Introduction mechanisms.
+	</p>
+	<p class="ednote" title="Minimum URL Requirements">Requiring the ability to read a TD from a URL over
+	HTTP would require even clients/consumers that only want to interact over CoAP, for instance, 
+	to implement an HTTP client to be able to support Discovery.
+	Another common way to configure a client/consumer is simply to read a (single) TD from a file.
+	There are also other valid ways to do discovery over CoAP, for example using CoRE-RD to discover
+	multiple TD links.
+	This is why in the following we allow any protocol to be used, even though
+	our Directory exploration mechanisms require HTTP.
+	</p>
+	<p class="ednote" title="D-Consumer">
+	Currently the WoT Architecture specification does not clearly say that Consumers MUST support Discovery.  
+	This is the reason for the introduction of the D-Client and D-Consumer terms.
+	</p>
+	</p>
+	We will use the term <a>D-Consumer</a> to refer to a WoT Consumer supporting WoT Discovery,
+	and D-Client as a more general term relating to any client supporting WoT Discovery.
```

Comment:
The distinction between the two terms is not clear. I'm really not sure we need the two terms.

Reply:
The components of a system can be named arbitrarily. I can argue that I need a Discovery Parser which parses the TD object incrementally. I can also argue that a TD is too large to pass around and should be transported and consumed incrementally (we have assertions for these).

Reply:
Right, the problem is that "Discoverer" sounds like "an entity capable of doing Discovery" which is not, however, its definition.  We could
1. Use "Discoverer" rather than "D-Consumer" and "D-Client" and use another term for the current meaning (e.g. "Registerer")
2. Leave it as is.
3. Get rid of D-Client (since formally, since D-Clients need to process TDs, they are in fact D-Consumers).  However, definition of Consumer implies interactions with Things, including endpoint Things (not just directories, so @k-toumura would prefer different terms).

Consensus: do 1, and also use this to resolve 3 (say a Discoverer MAY be a Consumer, but not necessarily).